improver/calibration/reliability_calibration.py

```diff
@@ -1268,10 +1268,19 @@ def _interpolate(
 
         forecast_probabilities = np.ma.getdata(forecast_threshold).flatten()
 
+        # interpolate using scipy first to get extrapolated values at endpoints
         interpolation_function = scipy.interpolate.interp1d(
             reliability_probabilities, observation_frequencies, fill_value="extrapolate"
         )
-        interpolated = interpolation_function(forecast_probabilities.data)
+        y_0, y_1 = interpolation_function([0, 1])
+        xp = np.copy(reliability_probabilities)
+        xp[0] = 0
+        xp[-1] = 1
+        fp = np.copy(observation_frequencies)
+        fp[0] = y_0
+        fp[-1] = y_1
```

Comment:
Should we be overwriting the end values here in the inputs? Would it make more sense to append values?

Reply:
This is correct as is, but is non-obvious and it took me a while to remember how it worked! Added a comment to clarify.